ui/src/components/build/create_build_config/SourceSection.js

```diff
@@ -0,0 +1,127 @@
+          selected={authType}
+        />
+      </FormGroup>
+      {authType === BUILD_AUTH_TYPE_BASIC ? renderBasicAuthentication(basicAuthState) : <React.Fragment />}
+      {authType === BUILD_AUTH_TYPE_SSH ? renderSshAuthentication(sshAuthState) : <React.Fragment />}
+    </div>
+  );
+}
+
+export function renderBasicAuthentication(basicAuthState) {
+  const { name = '', username = '', password = '' } = basicAuthState.getOrEmpty();
+  return (
+    <React.Fragment>
+      <FormGroup validationState={basicAuthState.getValidationState(KEY_CR_BASIC_AUTH_NAME)}>
+        <ControlLabel className="required">Name</ControlLabel>
+        <FormControl
+          type="text"
+          onChange={e => basicAuthState.set(KEY_CR_BASIC_AUTH_NAME, e.target.value)}
+          value={name}
+        />
+        <HelpBlock>A name for the credentials.</HelpBlock>
+      </FormGroup>
+      <FormGroup validationState={basicAuthState.getValidationState(KEY_CR_BASIC_AUTH_USERNAME)}>
+        <ControlLabel className="required">Username</ControlLabel>
+        <FormControl
+          type="text"
+          onChange={e => basicAuthState.set(KEY_CR_BASIC_AUTH_USERNAME, e.target.value)}
+          value={username}
+        />
+        <HelpBlock>Username for Git authentication.</HelpBlock>
+      </FormGroup>
+      <FormGroup validationState={basicAuthState.getValidationState(KEY_CR_BASIC_AUTH_PASSWORD)}>
+        <ControlLabel className="required">Password or Token</ControlLabel>
+        <FormControl
+          type="password"
+          onChange={e => basicAuthState.set(KEY_CR_BASIC_AUTH_PASSWORD, e.target.value)}
+          value={password}
+        />
+        <HelpBlock>Password or token for Git authentication.</HelpBlock>
+      </FormGroup>
+    </React.Fragment>
+  );
+}
+
+export function renderSshAuthentication(sshAuthState) {
+  const { name = '', privateKey = '' } = sshAuthState.getOrEmpty();
+  return (
+    <React.Fragment>
+      <FormGroup validationState={sshAuthState.getValidationState(KEY_CR_SSH_AUTH_NAME)}>
+        <ControlLabel className="required">Name</ControlLabel>
+        <FormControl type="text" onChange={e => sshAuthState.set(KEY_CR_SSH_AUTH_NAME, e.target.value)} value={name} />
+        <HelpBlock>A name for the credentials.</HelpBlock>
+      </FormGroup>
+      <FormGroup validationState={sshAuthState.getValidationState(KEY_CR_SSH_PRIVATE_KEY)}>
+        <ControlLabel className="required">SSH Private Key</ControlLabel>
+        <UploadControl onTextLoaded={text => sshAuthState.set(KEY_CR_SSH_PRIVATE_KEY, text)} />
+        <HelpBlock>Upload your private SSH key file.</HelpBlock>
+        <FormControl
+          componentClass="textarea"
+          onChange={e => sshAuthState.set(KEY_CR_SSH_PRIVATE_KEY, e.target.value)}
+          value={privateKey}
+        />
+        <HelpBlock>Private SSH key file for Git authentication.</HelpBlock>
+      </FormGroup>
+    </React.Fragment>
+  );
```

Comment:
I think these should really be react components

Reply:
agree with @wei-lee